frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-install.tsx

```diff
@@ -0,0 +1,304 @@
+            <button type="button" className="btn btn-default" onClick={history.goBack}>
+              Cancel
+            </button>
+          </ButtonBar>
+        </form>
+      </div>
+    );
+  }
+}
+
+export const CreateOCSServiceYAML: React.FC<CreateOCSServiceYAMLProps> = (props) => {
+  const template = _.attempt(() => safeDump(props.sample));
+  if (_.isError(template)) {
+    // eslint-disable-next-line no-console
+    console.error('Error parsing example JSON from annotation. Falling back to default.');
+  }
+
+  return <CreateYAML template={!_.isError(template) ? template : null} match={props.match} hideHeader={false} />;
+};
+
+/**
+ * Component which wraps the YAML editor and form together
+ */
+export class CreateOCSService extends React.PureComponent<CreateOCSServiceProps> {
+  state = {
+    sample: null,
+    method: 'form',
+    clusterServiceVersion: null,
+  };
+
+  componentDidMount() {
+    k8sGet(ClusterServiceVersionModel, this.props.match.params.appName, this.props.match.params.ns)
+      .then(clusterServiceVersionObj => {
+        this.setState({ sample: JSON.parse(_.get(clusterServiceVersionObj.metadata.annotations, 'alm-examples'))[0], clusterServiceVersion: clusterServiceVersionObj });
+      });
+  };
+
+  setMethod = (method) => {
+    this.setState({ method });
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <div className="co-create-operand__header">
+          <div className="co-create-operand__header-buttons">
+            {(this.state.clusterServiceVersion !== null) && <BreadCrumbs breadcrumbs={[
+              { name: this.state.clusterServiceVersion.spec.displayName, path: window.location.pathname.replace('/~new', '') },
+              { name: `Create ${OCSModel.label}`, path: window.location.pathname },
+            ]} />
+            }
+          </div>
+        </div>
+        {this.state.method === 'form' && <CreateOCSServiceForm
+          namespace={this.props.match.params.ns}
+          operandModel={OCSModel}
+          sample={this.state.sample}
+          clusterServiceVersion={this.state.clusterServiceVersion!== null && this.state.clusterServiceVersion.metadata}
+        /> ||
+          this.state.method === 'yaml' && <CreateOCSServiceYAML
+            match={this.props.match}
+            sample={this.state.sample}
+          />}
+      </React.Fragment>
+    );
+  }
+};
+
+type CreateOCSServiceProps = {
+  match: match<{ appName: string, ns: string, plural: K8sResourceKindReference }>;
+  operandModel: K8sKind;
+  sample?: K8sResourceKind;
+  namespace: string;
+  loadError?: any;
+  clusterServiceVersion: ClusterServiceVersionKind;
+};
+
+type CreateOCSServiceFormProps = {
+  operandModel: K8sKind;
+  sample?: K8sResourceKind;
+  namespace: string;
+  clusterServiceVersion: ClusterServiceVersionKind;
+};
+
+type CreateOCSServiceYAMLProps = {
+  sample?: K8sResourceKind;
+  match: match<{ appName: string, ns: string, plural: K8sResourceKindReference }>;
+};
```

Comment:
Nit: all source files (`ts`, `tsx`, `scss`, etc.) should have a trailing newline.